helix-core/src/main/java/org/apache/helix/messaging/DefaultMessagingService.java

```diff
@@ -186,7 +186,8 @@ private HelixDataAccessor getRecipientDataAccessor(final Criteria recipientCrite
       Map<String, String> sessionIdMap = new HashMap<String, String>();
       if (recipientCriteria.isSessionSpecific()) {
         Builder keyBuilder = targetDataAccessor.keyBuilder();
-        List<LiveInstance> liveInstances = targetDataAccessor.getChildValues(keyBuilder.liveInstances());
+        List<LiveInstance> liveInstances =
+            targetDataAccessor.getChildValues(keyBuilder.liveInstances(), true);
```

Comment:
Are we retrying for these? Or we need some way to monitor the failure ones.

Reply:
Retry is in https://github.com/apache/helix/pull/970.
I thought about this. If the liveInstance list is not complete, I guess we will only send partial messages, I guess it would be bad since there might be some relationship between the messages if they are done within the same pipeline.

But I don't have a strong preference. After PR 970 is in, there is very little chance we see this error. Please let me know if you think this shall be false (and why). I can change it and comment with the reason.

Reply:
I agree it will have little chance. Was worrying about if we throw an exception, it could totally miss the message delivery.

Reply:
I still not suggest to have this for DefaultMessaging service. Because this changes behavior and it is used by customer. It may break their logic.

Except this one, others are in our control and should be fine for me.

Reply:
Reasonable, I will change this to false.